app/services/IncomeSourceDetailsService.scala

```diff
@@ -76,4 +76,13 @@ class IncomeSourceDetailsService @Inject()(val businessDetailsConnector: Busines
       IncomeSourcesModel(businessList, property)
     }
   }
+
+  def getBusinessDetails(nino: String, selfEmploymentId: String)(implicit hc: HeaderCarrier): Future[Either[Option[BusinessModel],BusinessDetailsErrorModel]] = {
```

Comment:
The Type signature should be reversed. Traditionally for `Either[L,R]` the left-hand side is considered the 'failure' scenario. The right-hand side is the success scenario. 

Therefore the signature should be:
`Future[Either[BusinessDetailsErrorModel,Option[BusinessModel]]]`